tests/test_hypothesis.py

```diff
@@ -0,0 +1,88 @@
+import os
+from datetime import datetime, timezone
+from typing import Optional
+
+import pytest
+from hypothesis import given, settings, strategies
+from typing_extensions import TypedDict
+
+from pydantic_core import SchemaValidator, ValidationError
+
+settings_dict = {}
+hyp_max_examples = os.getenv('HYPOTHESIS_MAX_EXAMPLES')
+if hyp_max_examples:
+    settings_dict['max_examples'] = int(hyp_max_examples)
+    print(f'hypothesis settings: {settings_dict}')
```

Comment:
This seems like a better case for [settings profiles](https://hypothesis.readthedocs.io/en/latest/settings.html#settings-profiles) than explicit decorators.  I'd usually put that in `conftest.py`, but if you've only got one test module using Hypothesis then here is also fine.

Reply:
done, might be nice to have a informative repr for `settings` that should all settings or custom settings.